src/building/count.c

```diff
@@ -58,3 +143,145 @@ int building_count_upgraded(building_type type)
     }
     return upgraded;
 }
+
+int building_count_in_area(building_type type, int minx, int miny, int maxx, int maxy)
+{
+    int total = 0;
+    for (building *b = building_first_of_type(type); b; b = b->next_of_type) {
+        if ((b->state == BUILDING_STATE_IN_USE || b->state == BUILDING_STATE_CREATED) && b == building_main(b)) {
+            if (b->x >= minx && b->x <= maxx && b->y >= miny && b->y <= maxy) {
```

Comment:
I think you should use `map_building_at` instead. This way buildings that are partially in the area may not be counted.

Do note that this would be slightly more complex as you would have to keep track of which building id's have already been added, but it's doable.

Reply:
All right, I will make the change.
Probably going to take a while.
I don't have much free time at the moment.

Reply:
No problem, I also am very short on time right now. It shows as it took me 5 days to reply.

Reply:
Changes made.